sentinel/sentinel.go

```diff
@@ -126,24 +126,24 @@ func (p Proxy) handleActiveContract(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	providerPK, err := common.NewPubKey(parts[2])
+	spenderPK, err := common.NewPubKey(parts[2])
 	if err != nil {
-		p.logger.Error("fail to parse provider pubkey", "error", err, "pubkey", parts[2])
-		respondWithError(w, fmt.Sprintf("bad provider pubkey: %s", err), http.StatusBadRequest)
+		p.logger.Error("fail to parse spender pubkey", "error", err, "chain", parts[4])
+		respondWithError(w, "Invalid spender pubkey", http.StatusBadRequest)
 		return
 	}
 
-	chain, err := common.NewChain(parts[3])
+	providerPK, err := common.NewPubKey(parts[3])
 	if err != nil {
-		p.logger.Error("fail to parse chain", "error", err, "chain", parts[3])
+		p.logger.Error("fail to parse provider pubkey", "error", err, "pubkey", parts[2])
 		respondWithError(w, fmt.Sprintf("bad provider pubkey: %s", err), http.StatusBadRequest)
 		return
 	}
 
-	spenderPK, err := common.NewPubKey(parts[4])
+	chain, err := common.NewChain(parts[4])
 	if err != nil {
-		p.logger.Error("fail to parse spender pubkey", "error", err, "chain", parts[4])
-		respondWithError(w, "Invalid spender pubkey", http.StatusBadRequest)
+		p.logger.Error("fail to parse chain", "error", err, "chain", parts[3])
```

Comment:
this looks to be mistmatched. chain is parsing part[4] but the error is for parts[3]?

Reply:
https://github.com/arkeonetwork/arkeo/blob/00e0842b0e0059c0e63dc56f5ec31d82a29d17d1/tools/curleo/main.go#L114

Reply:
it's the leading slash

Reply:
yea, my bad.